### PR TITLE
8299865: Unnecessary NullPointerException catch in java.util.TimeZone#setDefaultZone

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
In [JDK-4368016](https://bugs.openjdk.org/browse/JDK-4368016) **TimeZone.setDefaultZone()** was updated so that 

_TimeZone.java [will] not cause an NPE. If getSystemTimeZoneID()
can't determine the time zone from java.home and user.region properties,
it returns GMT._ 

This change updates the code to get rid of the no longer needed Try Catch

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299865](https://bugs.openjdk.org/browse/JDK-8299865): Unnecessary NullPointerException catch in java.util.TimeZone#setDefaultZone

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * [Andrey Turbanov](https://openjdk.org/census#aturbanov) (@turbanoff - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11951/head:pull/11951` \
`$ git checkout pull/11951`

Update a local copy of the PR: \
`$ git checkout pull/11951` \
`$ git pull https://git.openjdk.org/jdk pull/11951/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11951`

View PR using the GUI difftool: \
`$ git pr show -t 11951`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11951.diff">https://git.openjdk.org/jdk/pull/11951.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/util/TimeZone.java [hunk 1]
```diff
@@ -675,14 +675,10 @@ private static synchronized TimeZone setDefaultZone() {
         // platform to Java time zone ID mapping.
         if (zoneID == null || zoneID.isEmpty()) {
             String javaHome = StaticProperty.javaHome();
-            try {
                 zoneID = getSystemTimeZoneID(javaHome);
```
Comment (naotoj) [inline src/java.base/share/classes/java/util/TimeZone.java:678]: Nit: align the indentation. Also, `javaHome` variable can be eliminated.

Comment (justin-curtis-lu) [reply src/java.base/share/classes/java/util/TimeZone.java]: Thanks for catching that, fixed

**Review by LanceAndersen — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by naotoj — APPROVED** [review]

**Review by turbanoff — APPROVED** [review]

**Pull request closed without merging**